src/js/Mixins/Snapping.js

```diff
@@ -333,6 +343,12 @@ const SnapMixin = {
     _getDistance(map, latlngA, latlngB) {
         return map.latLngToLayerPoint(latlngA).distanceTo(map.latLngToLayerPoint(latlngB));
     },
+    _middleLatLng(map, latlngA, latlngB) {
+        const p1 = map.project(latlngA);
+        const p2 = map.project(latlngB);
+
+        return map.unproject(p1._add(p2)._divideBy(2));
```

Comment:
ha, nice one!. Did not know `_divideBy(2)`

Reply:
are these private functions that we rather shouldn't use or would you say it's ok?

Reply:
In fact, I reused the method `_calcMiddleLatLng` (see [here](https://github.com/codeofsumit/leaflet.pm/blob/develop/src/js/Edit/L.PM.Edit.Line.js#L565)) in `L.PM.Edit.Line.js`. But with next commit, it will be moved into an "Utils.js" file.

About `_add()` and `_divideBy()` : these are used by public functions `add()` and `divideBy()` (see [Leaflet sources](https://github.com/Leaflet/Leaflet/blob/9fb0df7f0418243dd50ffaac715dc894a7afeb91/src/geometry/Point.js#L48)), but are destructive. That's fine here, because p1 and p2 are used only to calculate middle of segment. 😊